lib/uplinkc/testdata/require.h

```diff
@@ -25,3 +25,11 @@ do { \
         exit(1);\
     }\
 } while (0)
+
+#define require_error(err) \
+do { \
+    if(strcmp("", err) == 0) { \
+        printf("failed:\n\t%s:%d: %s\n", __FILE__, __LINE__, err);\
+        exit(1);\
+    }\
+} while (0)
```

Comment:
Out of curiosity, what `while(0)` does here?

Reply:
tbh don't know, maybe @bryanchriswhite will explain